lib/bundlex/toolchain/common/compilers.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Bundlex.Toolchain.Common.Compilers do
+  defstruct [:c, :cpp]
+
+  def resolve_lang(lang) do
+    cond do
+      lang in ["cpp", "c++", "C++", "CPP", "cplusplus", "CPLUSPLUS"] ->
+        :cpp
+
+      lang in ["c", "C"] ->
+        :c
+
+      true ->
+        :c
+    end
+  end
+
+  def resolve_compiler(lang, compilers) when is_atom(lang) do
+    with :cpp <- lang do
+      compilers.cpp
+    else
+      _lang -> compilers.c
```

Comment:
The atom here has to be `:c`, so it should be matched on instead of being ignored.